### PR TITLE
Extreme EXOS pack

### PR DESCRIPTION
**Pull request by davehammers**
This is an initial pack for Extreme EXOS switches.
Currently there is only one action which connects to an EXOS switch and runs a CLI command via JSONRPC http interface.

The pack config contains the username and password used to authenticate switch access. The EXOS defaults are username=admin, password='' <blank>

### CODE REVIEW

**Comment by LindsayHill** [comment]
I've created a new repo for this: https://github.com/StackStorm-Exchange/stackstorm-exos
You can open a PR against that when you're ready (or we can transfer it)

File: exos/actions/lib/jsonrpc.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        # first check the HTTP error code to see if HTTP was successful
+        # delivering the message
+        if response.status_code == requests.codes.ok:
+            # if we have a cookie, store it so we can use it later
+            self.cookie = response.cookies.get('session')
+            try:
+                # ensure the response is JSON encoded
+                jsonrpc_response = json.loads(response.text)
+            except:
+                raise ValueError("response not in JSON format")
```
Comment (Kami) [inline exos/actions/lib/jsonrpc.py]: Would be good to include response body here so it it's easier to debug if if we end up here.

File: exos/actions/lib/jsonrpc.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+                row_lines = cli_output.splitlines()
+                for line in row_lines:
+                    lower_case_line = line.lower()
+                    if lower_case_line.startswith('run time error'):
+                        raise RuntimeError(line)
+                    if lower_case_line.startswith('error'):
+                        raise RuntimeError(line)
+        else:
+            raise ValueError("Unexpected JSON result data format")
```
Comment (Kami) [inline exos/actions/lib/jsonrpc.py]: Same here - probably a good idea to include raw response body here to make it easier to debug.

**Comment by LindsayHill** [comment]
Closed, transferred to https://github.com/StackStorm-Exchange/stackstorm-exos/pull/1

**Pull request closed without merging**